onnx/defs/math/defs.cc

```diff
@@ -3087,30 +3088,56 @@ ONNX_OPERATOR_SET_SCHEMA(
             return;
           }
 
+          auto& input_shape = getInputShape(ctx, input_arg_index);
+
+          // The last dimension is the real and imaginary parts of the value.
+          const int64_t rank = input_shape.dim_size();
+          if (rank < 2) {
+            fail_shape_inference("input tensor must have rank >= 2, including the complex dimension.");
+          }
+
           // In general the output shape will match the input shape exactly
           // So initialize the output shape with the input shape
-          auto& input_shape = getInputShape(ctx, input_arg_index);
-          ONNX_NAMESPACE::TensorShapeProto result_shape_proto = input_shape;
+          TensorShapeProto result_shape_proto = input_shape;
+
+          // Handle when axis is an input but is not statically known
+          if (ctx.hasInput(axis_arg_index) && ctx.getInputData(axis_arg_index) == nullptr) {
+            // Axis is an input but is not statically known
+            if (is_onesided || ctx.hasInput(dft_length_arg_index)) {
+              // We don't know which axis is the signal dimension, so we cannot infer shape
+              // when onesided is enabled or when dft_length_arg_index is provided
+              result_shape_proto.clear_dim();
```

Comment:
Not sure about this. Instead of clearing dim, we should iterate through all but the last dimension, and clearing any dim_value or dim_param they have. Alternatively, it is probably simpler to construct a new TensorShapeProto, and add rank-1 dimensions (without setting anything) and adding the last dimension and setting it as below.

Reply:
Eg., see [this](https://github.com/onnx/onnx/blob/3abb6b3b38f1314688985824f915e893eab1cab2/onnx/defs/image/defs.cc#L64) and [this](https://github.com/onnx/onnx/blob/3abb6b3b38f1314688985824f915e893eab1cab2/tools/protobuf-bench.cc#L19)

Reply:
Done